File: main/java/com/example/wofertas/MainActivity.java
package com.example.wofertas;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.google.firebase.FirebaseApp;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class MainActivity extends AppCompatActivity {

    private FirebaseAuth auth;
    private FirebaseUser user;
    private Button btEntrar, btCadastrar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // Chama explicitamente a versão de AppCompatActivity
        super.setContentView(R.layout.activity_main);

        // Inicializa o Firebase
        FirebaseApp.initializeApp(this);
        auth = FirebaseHelper.getAuth();
        user = FirebaseHelper.getUsuarioAtual();
        btEntrar    = findViewById(R.id.button);
        btCadastrar  = findViewById(R.id.button2);

        if (user != null) {
            Log.d("Firebase", "Usuário logado: " + user.getEmail());
        } else {
            Log.d("Firebase", "Nenhum usuário logado.");
        }

        // Ajusta paddings para edge-to-edge (API 30+)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
        // Cadastro
        btEntrar.setOnClickListener(v ->
                startActivity(new Intent(MainActivity.this, LoginActivity.class))
        );

        // Voltar
        btCadastrar.setOnClickListener(v -> startActivity(new Intent(MainActivity.this,Cadastro.class))
        );
    }
}

File: app/src/main/java/com/example/wofertas/MainActivity.kt
package com.example.wofertas

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.work.Constraints
import androidx.work.NetworkType
import androidx.work.PeriodicWorkRequestBuilder
import androidx.work.WorkManager
import com.example.wofertas.fcm.NotificationHelper
import java.util.concurrent.TimeUnit

class MainActivity : AppCompatActivity() {

    private lateinit var btEntrar: Button
    private lateinit var btCadastrar: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // 1. Inicializa canais de notificação (Essencial para Android 8+)
        NotificationHelper.criarCanais(this)

        // 2. Verifica se já está logado
        if (AuthManager.isLoggedIn(this)) {
            // Se logado, garante que o buscador de ofertas em segundo plano está ativo
            agendarBuscaDeOfertas()
            redirecionarPorTipo()
            return
        }

        // 3. Configuração da Interface (Usuário Deslogado)
        btEntrar = findViewById(R.id.btn_entrar)
        btCadastrar = findViewById(R.id.btn_criar_conta)

        btEntrar.setOnClickListener {
            startActivity(Intent(this, LoginActivity::class.java))
        }
        btCadastrar.setOnClickListener {
            startActivity(Intent(this, Cadastro::class.java))
        }

        // Ajuste de layout para barras do sistema (status bar/navigation bar)
        val mainView = findViewById<android.view.View>(R.id.main)
        if (mainView != null) {
            ViewCompat.setOnApplyWindowInsetsListener(mainView) { v, insets ->
                val sb = insets.getInsets(WindowInsetsCompat.Type.systemBars())
                v.setPadding(sb.left, sb.top, sb.right, sb.bottom)
                insets
            }
        }
    }

    /**
     * Configura o WorkManager para buscar novas ofertas a cada 1 hora.
     * Só roda se houver internet para economizar bateria.
     */
    private fun agendarBuscaDeOfertas() {
        val constraints = Constraints.Builder()
            .setRequiredNetworkType(NetworkType.CONNECTED)
            .build()

        val request = PeriodicWorkRequestBuilder<OfertasWorker>(1, TimeUnit.HOURS)
            .setConstraints(constraints)
            .addTag(OfertasWorker.WORK_NAME)
            .build()

        WorkManager.getInstance(this).enqueueUniquePeriodicWork(
            OfertasWorker.WORK_NAME,
            androidx.work.ExistingPeriodicWorkPolicy.KEEP, // Mantém o agendamento se já existir
            request
        )
    }

    private fun redirecionarPorTipo() {
        val destino = when {
            AuthManager.isMercado(this) -> DashboardSupermercadoActivity::class.java
            AuthManager.isUsuario(this) -> ListaOfertas::class.java
            else -> {
                AuthManager.clearSession(this)
                LoginActivity::class.java
            }
        }

        startActivity(Intent(this, destino).apply {
            addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
        })
        finish()
    }
}
